regparser/web/jobs/models.py

```diff
@@ -25,3 +33,28 @@ class ParsingJob(models.Model):
         ("complete_with_errors", "complete_with_errors")
     ), default="received")
     url = models.URLField(blank=True, max_length=2000)
+
+    def save(self, *args, **kwargs):
+        super(ParsingJob, self).save(*args, **kwargs)
+
+
+class PipelineJob(ParsingJob):
+
+    cfr_title = models.IntegerField()
+    cfr_part = models.IntegerField()
+    only_latest = models.BooleanField(default=False)
+
+
+class ProposalPipelineJob(ParsingJob):
+
+    file_hexhash = models.CharField(max_length=32)
+    only_latest = models.BooleanField(default=True)
+
+
+class RegulationFile(models.Model):
+
+    contents = models.BinaryField()
+    file = models.FileField(null=True)
```

Comment:
(At some point I thought that storing the contents only in the DB might be the way to go, but I now don't think that's likely to hold up.)